pkgs/database/src/interface.zig

```diff
@@ -0,0 +1,51 @@
+const rocksdb = @import("rocksdb");
+const std = @import("std");
+
+/// Gets the return type of a function or function pointer
+pub fn ReturnType(comptime FnPtr: type) type {
+    return switch (@typeInfo(FnPtr)) {
+        .@"fn" => |fun| fun.return_type.?,
+        .pointer => |ptr| @typeInfo(ptr.child).@"fn".return_type.?,
+        else => @compileError("not a function or function pointer"),
+    };
+}
+
+/// A namespace for a column
+/// Can be used to iterate over a column
+/// and to find the index of a column family in a slice
+pub const ColumnNamespace = struct {
+    namespace: []const u8,
+    Key: type,
+    Value: type,
```

Comment:
why can't be Key/Value type comptime?

Reply:
I know you use `type`, my question is can it be  `comptime type`? and make `ColumnNamespace(comptime key: type, comptime value: type)`